src/mode/mode_struct.rs

```diff
@@ -1183,6 +1192,52 @@ impl<T: Transformation> Mode<T> {
         v
     }
 
+    fn is_in_sync(
+        &self,
+        jump_max: UnitValue,
+        time_expired: bool,
+        prev_value: UnitValue,
+        current_value: UnitValue,
+        target_value: UnitValue,
+        in_sync: bool,
+    ) -> bool {
+
+        let current_distance: f64 = current_value.get() - target_value.get();
+        let prev_distance: f64 = prev_value.get() - target_value.get();
+        let same_side: bool = (current_distance < 0.0 && prev_distance < 0.0) ||
```

Comment:
I think `f64` has  a `signum` method, so it should be enough here to compare the results of that method.